### PR TITLE
feat(be): fetch current param values from blockfrost

### LINKED ISSUE #15: Back end can return the list of current parameter values to the front end (use Blockfrost API)

### PR DESCRIPTION
**Pull request by bogdan-manole**

### CODE REVIEW

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 1]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+
+scalarX :: (ProtocolParams -> a) -> ProtocolParams -> Identity Integer
+scalarX = undefined
+
+allParamsWithCurrentValues :: [ParamWithCurrentValue]
+allParamsWithCurrentValues =
+  [ ParamWithCurrentValue
+      txFeePerByte
+      -- Not sure
+      (Identity . _protocolParamsMinFeeA)
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
      txFeePerByte
      (Identity . _protocolParamsMinFeeA)
```

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 2]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+  , --
+    ParamWithCurrentValue
+      txFeeFixed
+      -- Not sure
+      (Identity . _protocolParamsMinFeeB)
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
      txFeeFixed
      (Identity . _protocolParamsMinFeeB)
```

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 3]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+  , --
+    ParamWithCurrentValue
+      utxoCostPerByte
+      -- Not sure (fromIntegral ?????)
+      (Identity . fromIntegral . _protocolParamsCoinsPerUtxoWord)
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
      utxoCostPerByte
      -- (fromIntegral ?????)
      (Identity . fromIntegral . _protocolParamsCoinsPerUtxoWord)
```

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 4]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+      (Identity . _protocolParamsCollateralPercent)
+  , --
+    ParamWithCurrentValue
+      maxCollateralInputs
+      -- OK
+      (Identity . _protocolParamsMaxCollateralInputs)
+  , --
+    ParamWithCurrentValue stakeAddressDeposit $
+      const unsure'
+  , --
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
    ParamWithCurrentValue stakeAddressDeposit $
    -- fromIntegral?????
      (Identity . fromIntegral .  _protocolParamsKeyDeposit)
  , --
```

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 5]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+    ParamWithCurrentValue stakePoolDeposit $
+      const unsure'
+  , --
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
    ParamWithCurrentValue stakePoolDeposit $
    -- fromIntegral????
      (Identity . fromIntegral . _protocolParamsPoolDeposit)
  , --
```

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 6]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+    ParamWithCurrentValue poolRetireMaxEpoch $
+      const unsure'
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
    ParamWithCurrentValue poolRetireMaxEpoch $
     (Identity . _protocolParamsEMax)
```

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 7]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+  , ParamWithCurrentValue stakePoolTargetNum $
+      const unsure'
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
  , ParamWithCurrentValue stakePoolTargetNum $
      (Identity . _protocolParamsNOpt)
```

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 8]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+  , --
+    ParamWithCurrentValue committeeMinSize $
+      const unsure'
+  , --
+    ParamWithCurrentValue
+      monetaryExpansion
+      (Identity . MkRational . _protocolParamsRho)
+  , --
+    ParamWithCurrentValue treasuryCut $
+      const unsure'
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
    ParamWithCurrentValue treasuryCut $
      (Identity. MkRational . _protocolParamsTau)
```

File: backend/src/Cardano/Constitution/Checker/Params/Definition.hs [hunk 9]
```diff
@@ -496,34 +500,142 @@ committeeMinSize =
...
+      const $
+        replicate 5 unsure'
+  , --
+    ParamWithCurrentValue dRepVotingThresholds $
+      const $
+        replicate 10 unsure'
+  , --
+    ParamWithCurrentValue executionUnitPrices $ \ProtocolParams{..} ->
+      -- not sure
+      map MkRational [_protocolParamsPriceMem, _protocolParamsPriceStep]
```
Comment (RSoulatIOHK) [inline backend/src/Cardano/Constitution/Checker/Params/Definition.hs]:
```suggestion
    ParamWithCurrentValue executionUnitPrices $ \ProtocolParams{..} ->
      map MkRational [_protocolParamsPriceMem, _protocolParamsPriceStep]
```

File: backend/secrets/sanchonet.token
Comment (RSoulatIOHK) [inline backend/secrets/sanchonet.token:1]: Are you sure you want the API key in here?

**Review by RSoulatIOHK — APPROVED** [review]

File: backend/secrets/sanchonet.token
Comment (bogdan-manole) [reply backend/secrets/sanchonet.token:1]: at the end, I will erase it from my bf project, and add it to the .gitignore list.
But for the moment it's easier for everybody to just build and run the app

**Pull request merged**